### ISSUE TITLE
Feature: Speed up getNewBlockTransactions during createNewBlockTemplate

### ISSUE THREAD

**Issue by hairtail** [iron-fish/ironfish#2090]
### Description

I have run some tests on this.
With my powerful cpu 3960x, it takes: about 4 seconds with 300 transactions while it's about 200 ms with 10 transactions.
I think most miners have knowledge about this situation. So most of them mined almost empty blocks for higher profits while we have so many deposit transactions to be mined.

**Comment by hairtail** [comment]
Since we are testnet phase, there is an easy and experiment way to do this. We can make an extra bonus for good & registered big pool miners who mine blocks with as many transactions as they can. But, you know, it's not a good idea when mainnet.

**Comment by lwisne** [comment]
Yes, we are aware of this dynamic and have discussed among the core team. We want to make sure miners are incentivized to mine as many transactions a possible once it comes time for mainnet!
